sway/desktop/render.c

```diff
@@ -1863,54 +1781,22 @@ void output_render(struct sway_output *output, struct timespec *when,
 
 	int width, height;
 	wlr_output_transformed_resolution(wlr_output, &width, &height);
+
 	if (debug.damage == DAMAGE_RERENDER) {
 		pixman_region32_union_rect(damage, damage, 0, 0, width, height);
 	}
 
-	bool has_blur = false;
-	bool blur_optimize_should_render = false;
 	bool damage_not_empty = pixman_region32_not_empty(damage);
-	pixman_region32_t extended_damage;
-	pixman_region32_init(&extended_damage);
-	if (!fullscreen_con && !server.session_lock.locked && damage_not_empty) {
-		// Check if there are any windows to blur
-		struct workspace_effect_info effect_info = get_workspace_effect_info(output);
-		has_blur = effect_info.container_wants_blur || config->blur_enabled;
-		if (effect_info.should_render_optimized_blur) {
-			blur_optimize_should_render = true;
-			// Damage the whole output
-			pixman_region32_union_rect(damage, damage, 0, 0, width, height);
-		}
-
-		// Extend the damaged region
-		int expanded_size = effect_info.expanded_size;
-		if (expanded_size > 0) {
-			int32_t damage_width = damage->extents.x2 - damage->extents.x1;
-			int32_t damage_height = damage->extents.y2 - damage->extents.y1;
-			// Limit the damage extent to the size of the monitor to prevent overflow
-			if (damage_width > width || damage_height > height) {
-				pixman_region32_intersect_rect(damage, damage, 0, 0, width, height);
-			}
-
-			wlr_region_expand(damage, damage, expanded_size);
-			pixman_region32_copy(&extended_damage, damage);
-			wlr_region_expand(damage, damage, expanded_size);
-		} else {
-			pixman_region32_copy(&extended_damage, damage);
-		}
-	} else {
-		pixman_region32_copy(&extended_damage, damage);
-	}
-
-	if (debug.damage == DAMAGE_HIGHLIGHT && damage_not_empty) {
-		fx_renderer_clear((float[]){1, 1, 0, 1});
-	}
 
 	if (!damage_not_empty) {
 		// Output isn't damaged but needs buffer swap
 		goto renderer_end;
 	}
 
+	if (debug.damage == DAMAGE_HIGHLIGHT) {
+		fx_renderer_clear((float[]){1, 1, 0, 1});
+	}
+
```

Comment:
We also have to make sure that both `-D damage=highlight` and `-D damage=rerender` works as expected

Reply:
logic shouldnt have changed but good callout! Best to be safe

Reply:
Seems like `highlight` is broken in master so might as well fix it here, mb 😅